solo-io/gloo
# Split extauth apis

# Description
Split extauth API definitions into two files, using the current `extauth.proto` file for the public facing API and a new `extauth-internal.proto` file for the internal API.

## API changes
Moved extauth internal API definition from `extauth.proto` to `extauth-internal.proto`. While these changes will result in new files being generated, the overall contents/names/functions of the generated package should not change.

# Context
With both APIs in one file, it can be confusing as to which API you are looking at. By splitting the API definitions we make this clear.

## Testing steps
These changes have been pulled into solo-projects and pass CI: https://github.com/solo-io/solo-projects/pull/7145/commits (last commit on Nov 4 passed everything except `/validated-working` and changelog bot. Tests are no longer re-run after bulldozer merges).


# Checklist:
